electron/channels/compare-and-merge-diographs.js

```diff
@@ -1,6 +1,6 @@
 const { isEmpty } = require('../lib/utils')
 
-const MAXIMUM_SUBFOLDER_DEPTH = 15
+const MAXIMUM_SUBFOLDER_DEPTH = 8
```

Comment:
Circularity means that the diory has been already in the path. Could that be detected and thus prevent further propagation?

Reply:
Should this issue be added to some bug / improvement list?

Reply:
[UpdateFolder design paper](https://paper.dropbox.com/doc/Update-folder--BEO~oKXhebZYt84yS8eUXpEVAg-GmSxeADhey2BxFcUMtXms) has lots of notes about it... I think that will do it.